solana/pyth2wormhole/client/src/main.rs

```diff
@@ -171,301 +206,214 @@ fn handle_attest(
         .collect();
     let batch_count = batches.len();
 
-    // NOTE(2022-04-26): only increment this if `daemon` is false
-    let mut finished_count = 0;
-
-    // Stats
-    // TODO(2022-05-12): These should become Prometheus metrics in the future
-    let mut stats_start_time = Instant::now();
-    let mut tx_successes = 0;
-    let mut tx_send_failures = 0;
-    let mut tx_confirm_failures = 0;
-
-    let mut batch_wait_times = Duration::ZERO;
+    // Each future schedules individual attestation attempts per
+    // batch. Without daemon mode, we return after the first job is
+    // complete on the batch. In daemon mode, we never return and
+    // resend each batch continuously according to attestation conditions.
+    let attestation_sched_futs = batches.into_iter().map(|(batch_no, b)| {
+        let config4fut = config.clone();
+        let p2w_addr4fut = p2w_addr.clone();
+        let payer4fut = Keypair::from_bytes(&payer.to_bytes()).unwrap(); // Keypair has no clone
+        let rpc_url4fut = rpc_url.clone();
+        let commitment4fut = commitment.clone();
+        let batch_count4fut = batch_count;
```

Comment:
@Reisen can these be avoided in some way?

Reply:
I think the above only works because the internal reference is `'static`. Any non-copy by-value member would cause a `doesn't live long enough`. The read-only data I'm using isn't `'static`, so I'd probably need a bunch of `Arc`s. Not ideal, but may still be cheaper and cleaner than the clones. A regular attempt to borrow read-only  for `config` gives me this:
```rust
error[E0597]: `config` does not live long enough
   --> client/src/main.rs:216:27
    |
215 |       let attestation_sched_futs = batches.into_iter().map(|(batch_no, b)| {
    |                                                            --------------- value captured here
216 |           let config4fut = &config;
    |                             ^^^^^^ borrowed value does not live long enough
...
236 |                   let job = attestation_job(
    |  ___________________________-
237 | |                     rpc4job,
238 | |                     batch_no,
239 | |                     batch_count4fut,
...   |
245 | |                     sema.clone(),
246 | |                 );
    | |_________________- argument requires that `config` is borrowed for `'static`
...
363 |   }
    |   - `config` dropped here while still borrowed

```